hs_access_control/tests/test_communities.py

```diff
@@ -0,0 +1,646 @@
+        self.assertTrue(self.posts in self.dogs.gaccess.view_resources)
+        self.assertTrue(self.holes in self.cats.gaccess.view_resources)
+        self.assertTrue(self.holes not in self.dogs.gaccess.edit_resources)
+        self.assertTrue(self.posts not in self.cats.gaccess.edit_resources)
+        self.assertTrue(self.posts not in self.dogs.gaccess.edit_resources)
+        self.assertTrue(self.holes not in self.cats.gaccess.edit_resources)
+
+        # check that resources are found correctly in groups
+        self.assertTrue(self.posts in self.cats.gaccess.view_resources)
+        self.assertTrue(self.holes in self.dogs.gaccess.view_resources)
+
+        # reject ownership of community by a group
+        self.assertFalse(self.dog.uaccess.can_share_community_with_group(self.pets, self.dogs,
+                                                                         PrivilegeCodes.OWNER))
+        with self.assertRaises(PermissionDenied):
+            self.dog.uaccess.share_community_with_group(self.pets, self.dogs,
+                                                        PrivilegeCodes.OWNER)
+
+        # Privileges are unchanged by the previous act
+        self.assertEqual(self.pets.get_effective_group_privilege(self.dogs),
+                         PrivilegeCodes.VIEW)
+        self.assertEqual(self.pets.get_effective_group_privilege(self.cats),
+                         PrivilegeCodes.VIEW)
+
+        # upgrade share privilege
+        self.dog.uaccess.share_community_with_group(self.pets, self.dogs, PrivilegeCodes.CHANGE)
+
+        # privilege object created
+        ggp = GroupCommunityPrivilege.objects.get(group=self.dogs, community=self.pets)
+        self.assertEqual(ggp.privilege, PrivilegeCodes.CHANGE)
+
+        # provenance object created
+        ggp = GroupCommunityProvenance.objects.filter(group=self.dogs, community=self.pets)
+        self.assertEqual(ggp.count(), 2)
+
+        self.assertEqual(self.pets.get_effective_group_privilege(self.dogs),
+                         PrivilegeCodes.CHANGE)
+        self.assertEqual(self.pets.get_effective_group_privilege(self.cats),
+                         PrivilegeCodes.VIEW)
+
+        self.assertTrue(self.holes in self.dog.uaccess.view_resources)
+        self.assertTrue(self.holes in self.dog.uaccess.edit_resources)
+        self.assertTrue(self.holes in self.cat.uaccess.view_resources)
+        self.assertTrue(self.holes not in self.cat.uaccess.edit_resources)
+
+        # unshare community with group
+        self.assertTrue(self.dog.uaccess.can_unshare_community_with_group(self.pets, self.dogs))
+        self.dog.uaccess.unshare_community_with_group(self.pets, self.dogs)
+
+        self.assertEqual(self.pets.get_effective_group_privilege(self.dogs),
+                         PrivilegeCodes.NONE)
+        self.assertEqual(self.pets.get_effective_group_privilege(self.cats),
+                         PrivilegeCodes.VIEW)
+
```

Comment:
May also test what groups are in the community (pets) after unsharing group 'dogs'.

Reply:
Perhaps later after we've gotten the use case running.